frontend/web/src/features/hostdashboard/hostcalendar/components/CalendarGrid.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useRef, useState } from "react";
+import { toKey, isSameMonthUTC, dayNames, monthNames } from "../utils/date";
+import { cx } from "../utils/classNames";
+
+export default function CalendarGrid({
+  view,
+  cursor,
+  monthGrid,
+  selections,
+  prices,
+  bookingsByDate = {},
+  onToggle,
+  onDragSelect,
+}) {
+  // basic drag select (desktop)
+  const dragging = useRef(false);
+  const [range, setRange] = useState(new Set());
+  const [hoveredDate, setHoveredDate] = useState(null);
+
+  const handleMouseDown = (key) => {
+    dragging.current = true;
+    setRange(new Set([key]));
+  };
+  const handleEnter = (key) => {
+    if (!dragging.current) return;
+    setRange((prev) => new Set(prev).add(key));
+  };
+  const handleUp = () => {
+    if (dragging.current && range.size) onDragSelect([...range]);
+    dragging.current = false;
+    setRange(new Set());
+  };
+
+  if (view !== "month") {
+    return (
+      <div className="hc-calendar hc-calendar--placeholder">
+        {view === "week" ? "Week view (placeholder)" : "Day view (placeholder)"}
+      </div>
+    );
+  }
+
+  return (
+    <div className="hc-calendar" onMouseLeave={handleUp} onMouseUp={handleUp}>
+      <div className="hc-grid-head">
+        {dayNames.map((d) => (
+          <div key={d} className="hc-grid-head-cell">{d}</div>
```

Comment:
Name your parameter more descriptive i.e. 'day'


```suggestion
        {dayNames.map((day) => (
          <div key={day} className="hc-grid-head-cell">{day}</div>
```